fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreDeleteWhereTest.java

```diff
@@ -170,7 +185,294 @@ void testDeleteWhereGroupedCount() throws Exception {
+                            .mapToLong(savedRecord -> savedRecord.getHeader().getRecNo())
+                            .sum();
+                    assertEquals(expectedSum, getGroupedSum(sumIndex, Key.Evaluated.concatenate(path, num)));
+                    sumsByNum.put(num, expectedSum);
+                }
+                sumsByPathAndNum.put(path, sumsByNum);
+
+                long expectedSum = sumsByNum.values().stream()
+                        .mapToLong(Long::longValue)
+                        .sum();
+                assertEquals(expectedSum, getGroupedSumByPath(sumIndex, path));
+                sumsByPath.put(path, expectedSum);
+
+            }
+
+            commit(context);
+        }
+
+        // Delete a single path. This should work because header.path is a prefix of (though not the full prefix of)
+        // the key expression on which the index is defined
+        try (FDBRecordContext context = openContext()) {
+            openRecordWithHeader(context, hook);
+
+            final String pathToDelete = paths.get(random.nextInt(paths.size()));
+            recordStore.deleteRecordsWhere(Query.field("header").matches(Query.field("path").equalsValue(pathToDelete)));
+
+            // Assert that the deleted path is gone, but all other path sums should be the same
+            for (String path : paths) {
+                // Get the sum for each value of num
+                Map<Integer, Long> sumByNum = sumsByPathAndNum.get(path);
+                for (Map.Entry<Integer, Long> numAndSum : sumByNum.entrySet()) {
+                    long expectedSum = path.equals(pathToDelete) ? 0L : numAndSum.getValue();
+                    assertEquals(expectedSum, getGroupedSum(sumIndex, Key.Evaluated.concatenate(path, numAndSum.getKey())));
+                }
+
+                // Get the aggregate sum for an entire path
+                long expectedSum = path.equals(pathToDelete) ? 0L : sumsByPath.get(path);
+                assertEquals(expectedSum, getGroupedSumByPath(sumIndex, path));
+            }
+
+            // do not commit
+        }
+
+        // Try to delete in a way that would be inconsistent with the sum index
+        try (FDBRecordContext context = openContext()) {
+            openRecordWithHeader(context, hook);
+            final String pathToDelete = paths.get(random.nextInt(paths.size()));
+            Query.InvalidExpressionException err = assertThrows(Query.InvalidExpressionException.class,
+                    () -> recordStore.deleteRecordsWhere(Query.field("header").matches(Query.and(
+                            Query.field("path").equalsValue(pathToDelete),
+                            Query.field("rec_no").equalsValue(3L)))));
+
+            assertThat(err.getMessage(), containsString(String.format("deleteRecordsWhere not supported by index %s", sumIndex.getName())));
+        }
+    }
+
+    @Test
+    void testDeleteWhereRank() throws Exception {
```

Comment:
I think `rank` can have a groupedCount > 1, (not sure about other aggregates), it would probably be good to have a test of that.